libcalico-go/lib/ipam/ipam_block.go

```diff
@@ -393,6 +441,12 @@ func (b *allocationBlock) releaseByHandle(handleID string) int {
 	for o = 0; o < b.NumAddresses(); o++ {
 		// Only check allocated ordinals.
 		if b.Allocations[o] != nil && intInSlice(*b.Allocations[o], attrIndexes) {
+			if opts.SequenceNumber != nil && *opts.SequenceNumber != b.GetSequenceNumberForOrdinal(o) {
```

Comment:
This one is perhaps problematic because a handle can technically point to multiple addresses, which obviously will have their own sequence numbers.

I'm not sure what the right semantic is. Options are:
- Prevent passing of SequenceNumbers on ReleaseByHandle calls (just error)
- Only release the allocations that match the handle and the sequence number, ignore others.
- Assert that all the IPs allocated with that handle have the given sequence number, fail if not.